clang/include/clang/Frontend/ASTUnit.h

```diff
@@ -729,16 +729,15 @@ class ASTUnit {
   /// \returns - The initialized ASTUnit or null if the AST failed to load.
   static std::unique_ptr<ASTUnit> LoadFromASTFile(
       StringRef Filename, const PCHContainerReader &PCHContainerRdr,
-      WhatToLoad ToLoad, std::shared_ptr<DiagnosticOptions> DiagOpts,
+      WhatToLoad ToLoad, IntrusiveRefCntPtr<llvm::vfs::FileSystem> VFS,
```

Comment:
I feel like it is better to pass `vfs::FileSystem&` to avoid ref counting? The caller should hold onto VFS during the function call and no ownership is transferred here.

Reply:
I agree in principle, but `ASTUnit` is special when it comes to lifetimes and actually typically does outlive the caller. Moreover, the `VFS` is used to create a `FileManager` that currently does take the ownership.